### ISSUE TITLE
Compromise-dates plugin: Support duration ranges

### ISSUE THREAD

**Issue by Fdawgs** [spencermountain/compromise#1135]
At present, the compromise-dates plugin supports `date-ranges` such as `aug 1 - aug 31`.

It would be great if it also supported duration ranges and project those accordingly, for example:
- `8 - 10 weeks`
- `1 to 2 days`

As an example:
```js
'use strict'

/** @type {import('compromise').default} */
const nlp = require('compromise')
const nlpDates = require('compromise-dates')
nlp.plugin(nlpDates)

const text = 'I will see him in 8-10 weeks'

const doc = nlp(text)

const context = {
  today: '2023-01-01',
}

const datesViews = doc.dates(context)
const dateJson = datesViews.get()[0]

console.log(dateJson)
// Would expect this to output:
// {
//   start: '2023-02-26T00:00:00.000Z',
//   end: '2023-03-12T23:59:59.999Z',
//   timezone: 'Europe/London',
//   duration: { years: 0, months: 0, days: 14, hours: 0, minutes: 0 }
// }
```

I'm happy to have a crack at this myself but not sure where to start?

**Comment by spencermountain** [comment]
love this - i think it should be pretty doable to add this template to [here](https://github.com/spencermountain/compromise/blob/master/plugins/dates/src/api/parse/range/02-date-range.js)- 
you could just put-together a match for it, like:  `in [<min>#Value] to [<max>#Value] [<unit>(weeks|months|days)]` or something like that, and then shift it by these values.

You can see something similar [here](https://github.com/spencermountain/compromise/blob/master/plugins/dates/src/api/parse/range/03-one-date.js#L40) - when it gets the 'today' date with `let start = new Unit(context.today)` and then adds the units with `start.applyShift({ weeks: -2 })`.

Let me know if I can point any closer, there's a lot i'd like to add in here, too.

The only issue I can think of maybe an order issue, if it rubbed-up against dates like `'8-10 weeks after June` or something? Should be fine!
Thanks in advance for your help.
cheers

**Comment by spencermountain** [comment]
feature released in 14.14.1
